src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1436,8 +1440,19 @@ public void addWayPoint(final BlockPos point, IBlockState block)
      * @return Map with ID (coordinates) as key, and buildings as value.
      */
     @NotNull
-    public Map<BlockPos, AbstractBuilding> getBuildings()
+    public ImmutableMap<BlockPos, IBuilding> getBuildings()
```

Comment:
Why would it not Need to be an Immutable. Do you want something from outside your colony to mess up the map, ofcourse not. If it would not be an immutable, somebody might remove a Building and the request System does not get notified.